docs/basics/mutations.md

```diff
@@ -92,6 +92,23 @@ const Todo = ({ id, title }) => {
 };
 ```
 
+### Adding typenames
```

Comment:
I think this is a great section to have. I wonder if it fits in `basics` though. Not really sure if there's anywhere else it could go?

Reply:
I think it should go into "Document Caching". We already have a gotchas section there. Or maybe we want to have the actual API here in `mutations`, note that it's for "Document Caching" only, then repeat the generic API in "Document Caching"?

Reply:
I mean this can be used for multiple things, think about localState, ... I wanted to move it here so we could give a real-world example.